ott/geometry/costs.py

```diff
@@ -25,6 +26,37 @@
 from ott.geometry import matrix_square_root
 
 
+def x_to_means_and_covs(x, dimension):
+  """Extract means and covariance matrices of Gaussians from raveled vector.
+
+  Args:
+    x: [num_gaussians, dimension, (1 + dimension)] jnp.ndarray of concatenated means and covariances (raveled)
+    dimension: the dimension of the Gaussians
+  Returns:
+    means: [num_gaussians, dimension] jnp.ndarray that holds the means.
+    covariances: [num_gaussians, dimension] jnp.ndarray that holds the covariances.
+  """
+  x = jnp.atleast_2d(x)
+  means = x[:, 0:dimension]
+  covariances = jnp.reshape(
+      x[:, dimension:dimension + dimension ** 2], (-1, dimension, dimension)
+  )
+  return jnp.squeeze(means), jnp.squeeze(covariances)
+
+
+def mean_and_cov_to_x(mean, covariance, dimension):
+  """Ravel a Gaussian's mean and covariance matrix to d(1 + d) vector."""
+  x = jnp.concatenate((mean, jnp.reshape(covariance, (dimension * dimension))))
+  return x
+
+
+means_and_covs_to_x = jax.vmap(mean_and_cov_to_x, in_axes=[0, 0, None])
```

Comment:
since this is used only once I would only consider adding it in the function it is called by. (and maybe remove this entirely since only used in the barycenter_init)

Reply:
yes that is a good point. 

The reason why I have kept both is really to be more in line with the fact that you have a preference for keeping these as utility functions and not methods of Bures. I have seen that both the vmaped and the non-vamped version can be useful, so I have kept both.

Again, I think this is a bit minor. I guess it could be either way